### ISSUE TITLE
Hara.reflect.common not conforming to ns spec

### ISSUE THREAD

**Issue by arichiardi** [zcaudate/hara#26]
Hello @zcaudate, I updated my project to `[org.clojure/clojure "1.9.0-alpha11"]` and including `ultra` causes this problem:

```
Exception in thread "main" java.lang.IllegalArgumentException: Call to clojure.core/ns did not conform to spec:
In: [1] val: ((require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string])) fails at: [:args] predicate: (cat :docstring (? string?) :attr-map (? map?) :clauses :clojure.core.specs/ns-clauses),  Extra input
:clojure.spec/args  (hara.reflect.pretty.classes (require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string]))
, compiling:(hara/reflect/pretty/classes.clj:1:1)
	at clojure.lang.Compiler.macroexpand1(Compiler.java:6795)
	at clojure.lang.Compiler.macroexpand(Compiler.java:6861)
	at clojure.lang.Compiler.eval(Compiler.java:6935)
	at clojure.lang.Compiler.load(Compiler.java:7403)
	at clojure.lang.RT.loadResourceScript(RT.java:374)
	at clojure.lang.RT.loadResourceScript(RT.java:365)
	at clojure.lang.RT.load(RT.java:455)
	at clojure.lang.RT.load(RT.java:421)
[...]
	at clojure.main$main.doInvoke(main.clj:385)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.IllegalArgumentException: Call to clojure.core/ns did not conform to spec:
In: [1] val: ((require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string])) fails at: [:args] predicate: (cat :docstring (? string?) :attr-map (? map?) :clauses :clojure.core.specs/ns-clauses),  Extra input
:clojure.spec/args  (hara.reflect.pretty.classes (require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string]))

	at clojure.spec$macroexpand_check.invokeStatic(spec.clj:627)
	at clojure.spec$macroexpand_check.invoke(spec.clj:616)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.lang.Compiler.macroexpand1(Compiler.java:6789)
	... 150 more
```

I guess this will require both `hara` and `ultra` to be bumped :smile: If I have a bit of time I will have a look myself and PR ok?

Thanks for your work!

Andrea.

**Comment by zcaudate** [comment]
wow. thanks for looking at this. 

spec definitely needs more readable outputs.

**Comment by arichiardi** [comment]
The error is in this `ns`: https://github.com/zcaudate/hara/blob/65fc70bfa9a4dc68eee8fce6e79ccbc2fe099b81/src/hara/reflect/pretty/classes.clj

But I cannot figure out what it is from:

```
xception in thread "main" clojure.lang.ExceptionInfo: Call to clojure.core/ns did not conform to spec:
In: [1] val: ((require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string])) fails at: [:args] predicate: (cat :docstring (? string?) :attr-map (? map?) :clauses :clojure.core.specs/ns-clauses),  Extra input
:clojure.spec/args  (hara.reflect.pretty.classes (require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string]))
 #:clojure.spec{:problems [{:path [:args], :reason "Extra input", :pred (cat :docstring (? string?) :attr-map (? map?) :clauses :clojure.core.specs/ns-clauses), :val ((require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string])), :via [], :in [1]}], :args (hara.reflect.pretty.classes (require [hara.reflect.common :refer :all] [hara.reflect.pretty.primitives :refer :all] [hara.common.string :as string]))}, compiling:(hara/reflect/pretty/classes.clj:1:1)
```

**Comment by arichiardi** [comment]
I have to add that by checking `ultra` I discovered that probably it is including some old version of `hara`: https://github.com/venantius/ultra/blob/0.4.1/project.clj

Moreover, the new `ultra` SNAPSHOT does not include `hara` anymore so maybe this issue does not belong here. Sorry for the noise :smile:

**Comment by zcaudate** [comment]
hmmm... I think I see the problem. the `require` has to be a keyword `:require`

I had a look at ultra before - not sure about the current version but it downloads it through it's own pomegranate calls.

**Comment by zcaudate** [comment]
I believe this has been fixed in the latest hara - 2.4.2
